tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_subvol_idempotence_earmark.py

```diff
@@ -0,0 +1,279 @@
+            # Set earmark
+            fs_util.set_subvolume_earmark(
+                client1, vol_name, subvol_name, earmark="nfs.share4"
+            )
+            # Get earmark
+            out = fs_util.get_subvolume_earmark(client1, vol_name, subvol_name)
+            if "nfs.share4" not in out:
+                log.error(f"Subvolume get earmark failed: {err}")
+                return 1
+            # Remove earmark
+            fs_util.remove_subvolume_earmark(client1, vol_name, subvol_name)
+            # Get earmark
+            out = fs_util.get_subvolume_earmark(client1, vol_name, subvol_name)
+            if "nfs.share4" in out:
+                log.error(f"Subvolume get earmark failed: {err}")
+                return 1
+            # Set earmark and rm earmark while IO is in progress
+            subvol_name = f"subvol15_{random_str}"
+            fs_util.create_subvolume(client1, vol_name, subvol_name)
+            # run IOs in the subvolume
+            sub_path, _ = client1.exec_command(
+                sudo=True, cmd=f"ceph fs subvolume getpath {vol_name} {subvol_name}"
+            )
+            with parallel() as p:
+                p.spawn(
+                    fs_util.run_ios(
+                        client1, f"{fuse_mounting_dir_1}{sub_path}", ["dd", "smallfile"]
+                    )
+                )
+            # Set earmark
+            fs_util.set_subvolume_earmark(
+                client1, vol_name, subvol_name, earmark="nfs.share5"
+            )
+            # Get earmark
+            out = fs_util.get_subvolume_earmark(client1, vol_name, subvol_name)
+            log.info(f"Get earmark output: {out}")
+            if "nfs.share5" not in out:
+                log.error(f"Subvolume get earmark failed: {err}")
+                return 1
+            # Remove earmark
+            fs_util.remove_subvolume_earmark(client1, vol_name, subvol_name)
+            # Get earmark
+            out = fs_util.get_subvolume_earmark(client1, vol_name, subvol_name)
+            if "nfs.share5" in out:
+                log.error(f"Subvolume get earmark failed: {err}")
+                return 1
+            log.info("Subvolume earmark operations successful")
+            return 0
+
+        return 0
+
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+
+    finally:
+        log.info("Cleaning up")
+        fs_util.client_clean_up(
+            "umount", fuse_clients=[clients[0]], mounting_dir=fuse_mounting_dir_1
+        )
+        # Clean up the subvolumes and volume
+        for i in range(1, 15):
+            subvol_name = f"subvol{i}_{random_str}"
+            client1.exec_command(
+                sudo=True,
+                cmd=f"ceph fs subvolume rm {vol_name}_{random_str} {subvol_name} --force || true",
+            )
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph fs volume rm {vol_name} --yes-i-really-mean-it || true",
```

Comment:
Can we also use below function to remove subvolumes, subvolume group and volumes where flag "i-really-mean-it" is handled

- def remove_subvolume
- def remove_subvolumegroup
- def remove_fs

Reply:
yes we can do that. let me apply that